### PR TITLE
update occupation columns to use last occurrence case for occupation

### PR DESCRIPTION
**Pull request by mkangia**

### CODE REVIEW

File: custom/enikshay/management/commands/data_dumps_person_case.py [hunk 1]
```diff
@@ -44,6 +44,15 @@ def get_last_episode(self, case):
             return Exception("could not find last episode for person %s" % case.case_id)
         return self.context['last_episode']
 
+    def get_last_occurrence(self, case):
+        self.context['last_occurrence'] = (
+            self.context.get('last_occurrence') or
+            get_last_occurrence(case)
+        )
+        if not self.context['last_occurrence']:
+            return Exception("could not find last occurrence for person %s" % case.case_id)
```
Comment (esoergel) [inline custom/enikshay/management/commands/data_dumps_person_case.py]: This should be `raise`d

File: custom/enikshay/management/commands/data_dumps_person_case.py [hunk 2]
```diff
@@ -105,6 +119,39 @@ def get_all_episode_cases_from_person(domain, person_case_id):
...
+    return CaseAccessors(domain).get_reverse_indexed_cases(
+        [occurrence_case_id], case_types=[CASE_TYPE_EPISODE])
+
+
+def get_last_occurrence(person_case):
+    occurrence_cases = get_all_occurrence_cases_from_person(person_case.domain, person_case.case_id)
+    open_occurrence_cases = [
+        occurrence_case for occurrence_case in occurrence_cases
+        if not occurrence_case.closed
+    ]
```
Comment (esoergel) [inline custom/enikshay/management/commands/data_dumps_person_case.py]: Feel free to punt on this, but FYI, `get_reverse_indexed_cases` also accepts an `is_closed` argument, if you wanna avoid doing this in memory.

File: custom/enikshay/management/commands/data_dumps_person_case.py [hunk 3]
```diff
@@ -105,6 +119,39 @@ def get_all_episode_cases_from_person(domain, person_case_id):
...
+        raise Exception("Multiple open occurrence cases for person %s" % person_case.case_id)
+    elif len(open_occurrence_cases) == 1:
+        return open_occurrence_cases[0]
+    else:
+        # dict to maintain episodes ids to occurrence case mapping to avoid fetching occurrence again
+        # later from latest episode case
+        episode_id_to_occurrence = {}
+        all_episode_cases = []
+        for occurrence_case in occurrence_cases:
+            episodes = get_all_episode_cases_from_occurrence(occurrence_case.domain, occurrence_case.case_id)
```
Comment (esoergel) [inline custom/enikshay/management/commands/data_dumps_person_case.py]: nit: you can pass the list of occurrence case ids to `get_reverse_indexed_cases` to fetch all the episodes in one db call.

Comment (esoergel) [reply custom/enikshay/management/commands/data_dumps_person_case.py]: oh, I guess you need to know which occurrence it came from, so you'd have to split that back out anyways, maybe not worth it.

**Review by esoergel — CHANGES_REQUESTED** [review]
`return` vs `raise` is the only blocker, otherwise looks g2g.

File: custom/enikshay/management/commands/data_dumps_person_case.py (hunk 1 above)
Comment (mkangia) [reply custom/enikshay/management/commands/data_dumps_person_case.py]: oh no.. dont understand why i keep doing this.. sorry.. will update..

File: custom/enikshay/management/commands/data_dumps_person_case.py (hunk 2 above)
Comment (mkangia) [reply custom/enikshay/management/commands/data_dumps_person_case.py]: Need all occurrence_cases in there are no open occurrence cases so i think fine to not do that here.

**Comment by mkangia** [comment]
thanks for the review @esoergel .. updated and replied.

**Review by esoergel — APPROVED** [review]

**Pull request merged**